### PR TITLE
Move off release candidate versions for `webpack-dev-server` and `react-refresh-webpack-plugin`

### PR DESCRIPTION
**Pull request by askoufis**
The main update is moving off the `webpack-dev-server` 4.0 release candidate. This didn't actually break anything, but some config has been deprecated between `4.0.0` and `4.11.0`, so I've moved to the new config.

The updates to `react-refresh-webpack-plugin` and `react-refresh` required no additional changes. This PR basically replaces #643.

I also added an `onDeath` handler to `sku start`. `sku start-ssr` has one, figured it's worth adding it to `sku start` as well.

### CODE REVIEW

File: .changeset/nervous-dragons-marry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'sku': patch
+---
+
+Update deps
```
Comment (askoufis) [inline .changeset/nervous-dragons-marry.md:5]: Adding this changeset now, but going to do a few more dependency update PRs before actually releasing it.

File: scripts/start.js [hunk 2]
```diff
@@ -179,4 +194,10 @@ const hot = process.env.SKU_HOT !== 'false';
 
     openBrowser(url);
   });
+
+  onDeath(() => {
+    devServer.stopCallback(() => {
+      debug('Webpack dev server closed');
+    });
+  });
```
Comment (askoufis) [inline scripts/start.js:202]: Just found https://webpack.js.org/configuration/dev-server/#devserversetupexitsignals, might replace this and the same bit in `start-ssr` with that config instead.

**Review by mrm007 — APPROVED** [review]

File: scripts/start.js [hunk 3]
```diff
@@ -94,76 +97,88 @@ const hot = process.env.SKU_HOT !== 'false';
...
-        }
-      }
-      app.get('*', (req, res, next) => {
-        const matchingSiteName = getSiteForHost(req.hostname);
-
-        const matchingRoute = routes.find(({ route, siteIndex }) => {
+  const devServer = new WebpackDevServer(
+    {
+      ...devServerConfig,
+      setupMiddlewares: (middlewares, { app }) => {
```
[the remark this answers is not in the document]
Comment (askoufis) [reply scripts/start.js]: Lost the types when I moved `setupMiddlewares` inside `devServerConfig`, so I added a type annotation. Thoughts? https://github.com/seek-oss/sku/pull/706/commits/b760e6eea641271cda42ca4cd65586a5da13362e

[the remark this answers is not in the document]
Comment (mrm007) [reply scripts/start.js]: Very nice! 👏

**Pull request merged**